### PR TITLE
Remove peer approval checks and add enabled field to control interface.

### PR DESCRIPTION
**Pull request by Osmose**
Once this is merged we should create a branch for peer approval that reverts this commit.

@rehandalal r?

### CODE REVIEW

File: normandy/control/static/control/admin/sass/control.scss [hunk 1]
```diff
@@ -114,7 +114,10 @@
 .crud-form {
   padding: 30px 45px;
 
-  input[type="checkbox"] { opacity: 0; }
+  input[type="checkbox"] {
+    width: auto;
+    margin: 0 5px;
+  }
```
Comment (rehandalal) [inline normandy/control/static/control/admin/sass/control.scss]: nit: blank line here

File: normandy/recipes/models.py [hunk 2]
```diff
@@ -119,22 +118,10 @@ def __str__(self):
...
-                if field in dirty_fields:
-                    dirty_fields.pop(field)
-            if dirty_fields:
-                self.disable()
-
-            # Increment the revision ID
-            self.revision_id += 1
-
-        super(Recipe, self).save(*args, **kwargs)
+        # Increment the revision ID
```
Comment (rehandalal) [inline normandy/recipes/models.py]: probably want to make this:
```python
if self.is_dirty(check_relationship=True):
    self.revision_id += 1
```

File: normandy/recipes/tests/test_api.py [hunk 3]
```diff
@@ -115,10 +115,13 @@ def test_it_serves_recipes(self, api_client):
...
-                                                  'action_name': action.name,
-                                                  'arguments': {},
-                                                  'filter_expression': 'whatever'})
+        res = api_client.post('/api/v1/recipe/', {
+            'name': 'Test Recipe',
+            'action_name': action.name,
+            'arguments': {},
+            'filter_expression': 'whatever',
+            'enabled': True
+        })
```
Comment (mythmon) [inline normandy/recipes/tests/test_api.py]: :+1:

File: normandy/recipes/tests/test_api.py [hunk 4]
```diff
@@ -208,12 +211,14 @@ def test_it_can_enable_recipes(self, api_client):
...
         assert recipe.enabled
 
+    @pytest.mark.xfail
     def test_cannot_enable_unapproved_recipes(self, api_client):
         recipe = RecipeFactory(approval=None, enabled=False)
 
         res = api_client.post('/api/v1/recipe/%s/enable/' % recipe.id)
         assert res.status_code == 400
 
+    @pytest.mark.xfail(strict=False)
```
Comment (mythmon) [inline normandy/recipes/tests/test_api.py]: Why is this `strict=False` and the other is not?

File: normandy/recipes/tests/test_api.py [hunk 5]
```diff
@@ -208,12 +211,14 @@ def test_it_can_enable_recipes(self, api_client):
         recipe = Recipe.objects.all()[0]
         assert recipe.enabled
 
+    @pytest.mark.xfail
```
Comment (mythmon) [inline normandy/recipes/tests/test_api.py]: `xfail` takes an optional `reason` kwarg. I think we should use that for all xfails.

File: normandy/control/static/control/js/components/form_fields/CheckboxField.jsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+const CheckboxField = (props) => {
```
Comment (mythmon) [inline normandy/control/static/control/js/components/form_fields/CheckboxField.jsx]: I'm not terribly excited about the `const Foo = (args) => { }` style of function declaration. This is a style nit though, so feel free to ignore this.

File: normandy/recipes/tests/test_api.py [hunk 7]
```diff
@@ -212,12 +215,14 @@ def test_it_can_enable_recipes(self, api_client):
...
         assert recipe.enabled
 
+    @pytest.mark.xfail(reason='disabled approvals temporarily')
     def test_cannot_enable_unapproved_recipes(self, api_client):
         recipe = RecipeFactory(approval=None, enabled=False)
 
         res = api_client.post('/api/v1/recipe/%s/enable/' % recipe.id)
         assert res.status_code == 400
 
+    @pytest.mark.skip(reason='disabled approvals temporarily')
```
Comment (mythmon) [inline normandy/recipes/tests/test_api.py]: I don't think I'm happy with the idea of having unconditional skip tests. In sumo we had tests that were skipped in certain environments (like a developer without Elasticsearch). We made skips failures in CI. The only reason we don't do that in Normandy already is that I couldn't figure out how to do it.

I think we should either delete these tests, or leave them unmodified. They exist on the peer review branch. They don't provide any value here.

File: normandy/control/static/control/js/components/form_fields/CheckboxField.jsx (hunk 6 above)
Comment (Osmose) [inline normandy/control/static/control/js/components/form_fields/CheckboxField.jsx]: I was trying to be consistent with other parts of the codebase, but I too like a function declaration over this style. I'll change it and we'll see what happens when linting lands.

**Pull request merged**